libs/sarif/src/schema/tool.rs

```diff
@@ -0,0 +1,45 @@
+use alloc::borrow::Cow;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// The analysis tool that was run.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "camelCase")
+)]
+#[non_exhaustive]
+pub struct Tool {
+    /// The analysis tool that was run.
+    pub driver: ToolComponent,
+}
+
+impl Tool {
+    /// Create a new `Tool` with the given driver.
+    #[must_use]
+    pub const fn new(driver: ToolComponent) -> Self {
+        Self { driver }
+    }
+}
+
+/// A component, such as a plug-in or the driver, of the analysis tool that was run.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "camelCase")
+)]
+#[non_exhaustive]
+pub struct ToolComponent {
```

Comment:
`3.19.2`: At least one of `version` (`3.19.13`) and `semanticVersion` (`3.19.12`) **SHOULD** be present.